### PR TITLE
Feature/update push per specs update

### PR DESCRIPTION
**Pull request by TalhaAli00**

### CODE REVIEW

File: sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt [hunk 1]
```diff
@@ -54,7 +54,17 @@ interface SignInterface {
...
+     * Caution: This function is blocking and runs on the current thread.
+     * It is advised that this function be called from background operation
+     */
     fun getListOfActiveSessions(): List<Sign.Model.Session>
+
+    /**
+     * Caution: This function is blocking and runs on the current thread.
+     * It is advised that this function be called from background operation
+     */
     fun getActiveSessionByTopic(topic: String): Sign.Model.Session?
```
Comment (Elyniss) [inline sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt:68]: should we add suspend?

**Review by Elyniss — APPROVED** [review]

File: sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt [hunk 2]
```diff
@@ -54,7 +54,17 @@ interface SignInterface {
     fun emit(emit: Sign.Params.Emit, onSuccess: (Sign.Params.Emit) -> Unit = {}, onError: (Sign.Model.Error) -> Unit)
     fun ping(ping: Sign.Params.Ping, sessionPing: Sign.Listeners.SessionPing? = null)
     fun disconnect(disconnect: Sign.Params.Disconnect, onSuccess: (Sign.Params.Disconnect) -> Unit = {}, onError: (Sign.Model.Error) -> Unit)
+
+    /**
+     * Caution: This function is blocking and runs on the current thread.
+     * It is advised that this function be called from background operation
+     */
     fun getListOfActiveSessions(): List<Sign.Model.Session>
```
[the remark this answers is not in the document]
Comment (TalhaAli00) [reply sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt:62]: Java does not support suspend functions so we can't

File: sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt (hunk 1 above)
Comment (TalhaAli00) [reply sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt:68]: same as above

File: sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakubuid) [reply sign/sdk/src/main/kotlin/com/walletconnect/sign/client/SignInterface.kt:62]: @TalhaAli00 I like the idea of detailed comments. Let's put them in all SDKs that expose such blocking methods (Core, Auth, W3W)

**Review by jakubuid — APPROVED** [review]

**Pull request merged**